src/runtime/contrib/random/mt_random_engine.cc

```diff
@@ -111,6 +115,56 @@ class RandomEngine {
     }
   }
 
+  void RandomFill(DLTensor* data) {
+    int64_t size = 1;
+    for (int i = 0; i < data->ndim; ++i) {
+      size *= data->shape[i];
+    }
+
+    if (data->ctx.device_type == kDLCPU) {
+      FillData(data, size);
+    } else {
+      DLTensor local;
```

Comment:
do not call raw allocate API, instead use NDArray::Empty here

Reply:
Updated.